optimizely/config_manager.py

```diff
@@ -0,0 +1,233 @@
+                 datafile=None,
+                 update_interval=None,
+                 url=None,
+                 url_template=None,
+                 logger=None,
+                 error_handler=None):
+        """ Initialize config manager. One of sdk_key or url has to be set to be able to use.
+
+        Args:
+          sdk_key: Optional string uniquely identifying the datafile.
+          update_interval: Optional floating point number representing time interval in seconds
+                           at which to request datafile and set ProjectConfig.
+          url: Optional string representing URL from where to fetch the datafile. If set it supersedes the sdk_key.
+          url_template: Optional string template which in conjunction with sdk_key
+                        determines URL from where to fetch the datafile.
+          logger: Provides a logger instance.
+          error_handler: Provides a handle_error method to handle exceptions.
+        """
+        super(PollingConfigManager, self).__init__(logger=logger, error_handler=error_handler)
+        self.datafile_url = self.get_datafile_url(sdk_key, url,
+                                                  url_template or enums.ConfigManager.DATAFILE_URL_TEMPLATE)
+        self.set_update_interval(update_interval)
+        self.last_modified = None
+        self._datafile = datafile
+        self._config = None
+        self._polling_thread = threading.Thread(target=self._run)
+        self._polling_thread.setDaemon(True)
+        if self._datafile:
+          self.set_config(self._datafile)
+
+    @staticmethod
+    def get_datafile_url(sdk_key, url, url_template):
+        """ Helper method to determine URL from where to fetch the datafile.
+
+        Args:
+          sdk_key: Key uniquely identifying the datafile.
+          url: String representing URL from which to fetch the datafile.
+          url_template: String representing template which is filled in with
+                        SDK key to determine URL from which to fetch the datafile.
+
+        Returns:
+          String representing URL to fetch datafile from.
+
+        Raises:
+          optimizely.exceptions.InvalidInputException if:
+          - One of sdk_key or url is not provided.
+          - url_template is invalid.
+        """
+        # Ensure that either is provided by the user.
+        if sdk_key is None and url is None:
+            raise optimizely_exceptions.InvalidInputException('Must provide at least one of sdk_key or url.')
+
+        # Return URL if one is provided or use template and SDK key to get it.
+        if url is None:
+            try:
+                return url_template.format(sdk_key=sdk_key)
+            except (AttributeError, KeyError):
+                raise optimizely_exceptions.InvalidInputException(
+                    'Invalid url_template {} provided.'.format(url_template))
+
+        return url
+
+    def set_update_interval(self, update_interval):
```

Comment:
I believe we're missing a docstring for `update_interval` here?

Reply:
Added